flask_resources/errors.py

```diff
@@ -33,6 +33,43 @@ def wrapper(self, *args, **kwargs):
     return wrapper
 
 
+class ErrorMapper:
+    """Class to map business logic errors to REST exceptions."""
+
+    exception_map = {}
+
+    def __init__(self, func):
+        """Constructor."""
+        update_wrapper(self, func)
+        self.func = func
+
+    def __call__(self, *args, **kwargs):
+        """Raise the mapped exception based on the `exception_map`."""
+        try:
+            return self.func(*args, **kwargs)
+        except tuple(self.exception_map.keys()) as exc:
+            exc_or_func = self.exception_map[exc.__class__]
+            if callable(exc_or_func) and not isinstance(exc_or_func, HTTPException):
+                exc_or_func = exc_or_func(exc)
+            raise exc_or_func from exc
+
+
+def error_mapper(f):
+    """Decorator that sets the error_mapper from the resource config."""
+
+    @wraps(f)
+    def wrapper(self, *args, **kwargs):
+        """Wrapping method.
+
+        :params self: Item/List/SingletonView instance
+        """
+        _error_mapper = getattr(self.resource.config, "error_mapper", None)
```

Comment:
If we put `error_mapper` on the default `ResourceConfig`, we won't need to use `getattr`.

Reply:
As this is configurable, I would say that is nice to not assume that the attribute is there :)

Reply:
I mean, if `error_mapper` is on `ResourceConfig`, then there is no way for it to not be defined (unless someone goes around `del` ing attributes but that's extremely unlikely and could be said about anything which would lead us to use `getattr` everywhere :smile: ...)